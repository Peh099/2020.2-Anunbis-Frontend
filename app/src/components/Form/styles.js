import styled from 'styled-components';

export const Container = styled.div`
        display: flex;
        align-items: center;
        flex-direction: column;
        height: fit-content;

        input {
            margin: 0px;
        }
    `;
export const Erro = styled.div`
    color: #F44336;
    font-size: 10px;
`;

export const Content = styled.form`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    `;

export const FooterContent = styled.div`
    width: fit-content;
    display: flex;
    align-items: center;
    justify-content: space-around;
    align-self:center;
    `;
