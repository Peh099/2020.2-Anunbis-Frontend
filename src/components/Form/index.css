.divImg {
    display: flex;
    box-sizing: border-box;
    border-radius: 15px;
    max-width: 500px;
    max-height: 500px;
    align-self: flex-end;
}

#imagem {
    width: 500px;
    height: 500px;
    background-image: url("../../assets/Logo.jpg");
    border-top-right-radius: 15px;
    border-bottom-right-radius: 15px;
}

.Form {
    display: flex;
    flex-direction: row-reverse;
    align-items: flex-start;
    background-color: #FFD54F;
    width: 1000px;
    height: 500px;
    max-width: 1000px;
    max-height: 500px;
    color: #000000;
    box-sizing: border-box;
    border-radius: 15px;
}

.Fields {
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    align-items: center;
    width: 500px;
    height: 300px;
}

.Button {
    display: flex;
    flex-direction: row;
    align-items: flex-start;
    align-items: center;
    text-align: center;
}

Button{
    background: #FFF9C4;
    border: 1px solid rgba(255, 245, 157, 0.6);
    box-sizing: border-box;
    border-radius: 10px;
    padding: 10px 5px;
    margin-inline: 10px;
}
