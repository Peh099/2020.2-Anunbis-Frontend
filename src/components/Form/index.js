import React from "react";
import Button from "../Button";
import './index.css';

export default function Form(props) {

    return (
        <div className="Form">
            <div className="divImg">
                <img id="imagem" />
            </div>
            <div className="Fields">
                <h4>
                    {props.title}
                </h4>
                <div>
                    {props.fields}
                </div>
                <div>
                    {props.buttons}
                </div>
            </div>
        </div>
    );
}