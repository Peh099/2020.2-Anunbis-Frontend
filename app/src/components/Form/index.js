import React from 'react';
import { FieldContainer, Erro, FormContainer, FooterContent } from './styles';

const Field = ({ errorMsg, margin, children }) => (
  <FieldContainer margin={margin}>
    {children}
    <Erro className="error">{errorMsg}</Erro>
  </FieldContainer>
);

const Footer = ({ children }) => <FooterContent>{children}</FooterContent>;

function Form({ children, onSubmit }) {
  return <FormContainer onSubmit={onSubmit}>{children}</FormContainer>;
}

Form.Field = Field;
Form.Footer = Footer;
export default Form;
