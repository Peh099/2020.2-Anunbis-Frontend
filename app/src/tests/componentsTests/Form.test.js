import renderer from 'react-test-renderer';
import Form from '../../components/Form';

describe('Snapshot Form component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<Form />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})