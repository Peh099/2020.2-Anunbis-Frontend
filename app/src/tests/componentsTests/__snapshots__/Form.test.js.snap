// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Form component matches the snapshot 1`] = `
<form
  className="sc-dlnjwi kCxWFn"
/>
`;
